Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7247,6 +7274,2082 @@ def panorama_install_file_content_update_command(args: dict):
+    Gets the operational information of the device groups in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    return PanoramaCommand.get_device_groups(topology, device_filter_string)
+
+
+def get_template_stacks(
+    topology: Topology, device_filter_string: Optional[str] = None
+) -> List[TemplateStackInformation]:
+    """
+    Gets the operational information of the template-stacks in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    return PanoramaCommand.get_template_stacks(topology, device_filter_string)
+
+
+def get_topology() -> Topology:
+    """
+    Builds and returns the Topology instance
+    """
+    server_url = demisto.params().get('server')
+    parsed_url = urlparse(server_url)
+    hostname = parsed_url.hostname
+    return Topology.build_from_string(
+        hostname,
+        username="",
+        password="",
+        api_key=demisto.params().get('key')
+    )
+
+
+def dataclasses_to_command_results(result: Any, empty_result_message: str = "No results."):
+    """
+    Given a dataclass or list of dataclasses,
+    convert it into a tabular format and finally return CommandResults to demisto.
+
+    :param empty_result_message: If the result data is non
+    """
+    if not result:
+        return CommandResults(
+            readable_output=empty_result_message,
+        )
+
+    # Convert the dataclasses into dicts
+    outputs: Union[list, dict] = {}
+    summary_list = []
+
+    if not hasattr(result, "summary_data"):
+        # If this isn't a regular summary/result return, but instead, is just one object or a list of flat
+        # objects
+        if isinstance(result, list):
+            outputs = [vars(x) for x in result]
+            summary_list = [vars(x) for x in result]
+            # This is a bit controversial
+            title = result[0]._title
+            output_prefix = result[0]._output_prefix
+        else:
+            outputs = vars(result)
+            summary_list = [vars(result)]
+            title = result._title
+            output_prefix = result._output_prefix
+    else:
+        if result.summary_data:
+            summary_list = [vars(x) for x in result.summary_data if hasattr(x, "__dict__")]
+            outputs = {
+                "Summary": summary_list,
+            }
+
+        if result.result_data:
+            outputs["Result"] = [vars(x) for x in result.result_data if hasattr(x, "__dict__")]  # type: ignore
+
+        title = result._title
+        output_prefix = result._output_prefix
+
+    extra_args = {}
+    if hasattr(result, "_outputs_key_field"):
+        extra_args["outputs_key_field"] = getattr(result, "_outputs_key_field")
+
+    readable_output = tableToMarkdown(title, summary_list)
```

Comment:
```suggestion
    readable_output = tableToMarkdown(title, summary_list, removeNull=True)
```

That will remove automatically empty columns if exist